### PR TITLE
updating normalized uri instead of saving the whole thing. we still have...

### PR DESCRIPTION
**Pull request by eishay**
... some save uri calls there and I changed the signature of update.

I would throw an exception if update fail then return "false"

### CODE REVIEW

**Comment by eishay** [comment]
@leogrim @raymondng

File: kifi-backend/modules/scraper/app/com/keepit/scraper/ShoeboxDbCallbacks.scala [hunk 1]
```diff
@@ -34,6 +34,16 @@ class ShoeboxDbCallbackHelper @Inject() (config:ScraperConfig, shoeboxServiceCli
       normalizedUriLock.unlock()
     }
   }
+  def syncUpdateNormalizedURIState(uriId: Id[NormalizedURI], state: State[NormalizedURI]): Unit = {
+    try {
+      normalizedUriLock.lock()
```
Comment (raymondng) [inline kifi-backend/modules/scraper/app/com/keepit/scraper/ShoeboxDbCallbacks.scala:39]: Do we still need this?

File: kifi-backend/modules/common/app/com/keepit/shoebox/ShoeboxServiceClient.scala [hunk 2]
```diff
@@ -614,12 +612,17 @@ class ShoeboxServiceClientImpl @Inject() (
...
 
   @deprecated("Dangerous call. Use updateNormalizedURI instead.","2014-01-30")
-  def saveNormalizedURI(uri:NormalizedURI): Future[NormalizedURI] = {
+  def saveNormalizedURI(uri: NormalizedURI): Future[NormalizedURI] = {
     call(Shoebox.internal.saveNormalizedURI(), Json.toJson(uri), callTimeouts = longTimeout, routingStrategy = leaderPriority).map { r =>
       r.json.as[NormalizedURI]
     }
   }
 
+  def updateNormalizedURIState(uriId: Id[NormalizedURI], state: State[NormalizedURI]): Future[Unit] = {
```
Comment (raymondng) [inline kifi-backend/modules/common/app/com/keepit/shoebox/ShoeboxServiceClient.scala:621]: :+1:

File: kifi-backend/modules/scraper/app/com/keepit/scraper/ShoeboxDbCallbacks.scala (hunk 1 above)
Comment (eishay) [inline kifi-backend/modules/scraper/app/com/keepit/scraper/ShoeboxDbCallbacks.scala:39]: I'm not sure. 
Once this code is alive I'll try to remove it

Comment (raymondng) [inline kifi-backend/modules/scraper/app/com/keepit/scraper/ShoeboxDbCallbacks.scala:39]: Cool

**Comment by raymondng** [comment]
Overall lgtm. Long overdue and thanks @eishay!

**Pull request merged**